phoenix-core/src/main/java/org/apache/phoenix/util/IndexUtil.java

```diff
@@ -294,13 +295,29 @@ private static boolean isEmptyKeyValue(PTable table, ColumnReference ref) {
                             .getLength()) == 0);
     }
 
+
+    public static boolean isGlobalIndexCheckerEnabled(PhoenixConnection connection, PName index)
+            throws SQLException {
+        TableDescriptor desc = desc = connection.getQueryServices().getTableDescriptor(index.getBytes());
```

Comment:
This is called from generateIndexData which will be high-traffic. What are the perf implications of grabbing the table descriptor very frequently, and do we need caching?

Looks like this creates an HTable each time which will in turn make a remote call to the Master each time
@gokceni @kadirozde

Reply:
adding to a static map.

Reply:
Actually I am not sure about caching this. If upgrade tool adds the coprocessor, map will go stale.